app/Model/Repository.php

```diff
@@ -51,4 +55,39 @@ public static function getViewers()
         }
         return $viewers;
     }
+
+    public static function factory(Project $project)
+    {
+        $service = null;
+        switch ($project->CvsViewerType) {
+            case self::VIEWER_GITHUB:
+                $token = Config::getInstance()->get('CDASH_GITHUB_API_TOKEN');
```

Comment:
This will need to be a per-project setting, not a global one. Ideas on where to store it:
`repositories.password` (and leave username blank?)
`apitoken` is really tempting, but that was meant for internal (not external) tokens
other ideas? security concerns?

Reply:
Yeah, I like that idea of keeping it in `repositories.password`.  I cannot think of any security concerns off of the top of my head... unless, I guess that table and its contents are restricted to users with project level read/write permissions? Other than that I think that definitely works.